Chat App/ViewModel/Pub/PubRegisterViewModel.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  PubRegisterViewModel.swift
+//  Chat App
+//
+//  Created by Amiel Jireh Cordova on 12/14/20.
+//  Copyright © 2020 Cordova, Jireh. All rights reserved.
+//
+
+import Foundation
+import RxCocoa
+import RxSwift
+
+protocol RegisterViewModelInputs {
+    func verifyUsernameAvailability(userInput: String)
+    func verifyUserInput(userInput: String) -> Bool
+    func verifyPasswordMatch(userInput: String!) -> Bool
+    func registerUser()
+}
+
+protocol RegisterViewModelOutputs {
+    var shouldShowAlert: PublishSubject<Bool> { get }
+    var shouldShowLoading: PublishSubject<Bool> { get }
+    var isUsernameAvailable: PublishSubject<Bool> { get }
+    var shouldProceedtoServer: PublishSubject<Bool> { get }
+}
+
+protocol RegisterViewModelTypes {
+    var inputs: RegisterViewModelInputs { get }
+    var outputs: RegisterViewModelOutputs { get }
+}
+
+class PubRegisterViewModel: RegisterViewModelInputs, RegisterViewModelOutputs, RegisterViewModelTypes {
+    var shouldShowAlert = PublishSubject<Bool>()
+    var shouldShowLoading = PublishSubject<Bool>()
+    var isUsernameAvailable = PublishSubject<Bool>()
+    var shouldProceedtoServer = PublishSubject<Bool>()
+    var inputs: RegisterViewModelInputs { return self }
+    var outputs: RegisterViewModelOutputs { return self }
+    
+    var alertTitle = ""
```

Comment:
internal use? maybe make it private instead?

Reply:
These properties are accessed from the viewmodel in the RegisterViewController.